audio-worklet/src/noisegate-script-processor.js

```diff
@@ -77,20 +79,30 @@ class NoiseGate {
    *                                     input and output buffers
    */
   onaudioprocess_(event) {
+    let inputBuffer = event.inputBuffer;
+    let channel0 = inputBuffer.getChannelData(0);
+
     // Stereo input is downmixed to mono input via averaging.
-    if (event.inputBuffer.numberOfChannels === 2) {
-      for (let i = 0; i < event.inputBuffer.length; i++) {
-        this.channel_[i] = (event.inputBuffer.getChannelData(0)[i] +
-                           event.inputBuffer.getChannelData(1)[i]) / 2;
+    if (inputBuffer.numberOfChannels === 2) {
+      let channel1 = inputBuffer.getChannelData(1);
+      for (let i = 0; i < channel1.length; i++) {
+        this.channel_[i] = (channel0[i] + channel1[i]) / 2;
       }
     } else {
-      this.channel_ = event.inputBuffer.getChannelData(0);
+      this.channel_ = channel0;
     }
 
     let envelope = this.detectLevel_(this.channel_);
 
-    for (let i = 0; i < event.inputBuffer.numberOfChannels; i++) {
-      let input = event.inputBuffer.getChannelData(i);
+    // For sine waves, the envelope eventually reaches an average power of
+    // a^2 / 2. Sine waves are therefore scaled back to the original amplitude,
```

Comment:
Did you look up or derive a^2/2?  I was only guessing. I didn't prove it or find a reference.
(You don't need to supply a reference or proof here. I'm just curious.)